File: Assignment 2/part1_syntax_semantics/type_scope_closure.cpp
#include <iostream>
#include <functional>
#include <string>
using namespace std;

int main() {
    string x = "Hello";
    cout << x << endl;

    function<void()> outer() {
        string x = "outer value";
        return [x]() { cout << "Accessing: " << x << endl; };
    }

    auto f = outer();
    f();

    return 0;
}

File: Assignment 2/part1_syntax_semantics/type_scope_closure.js
// Type system example
let x = 5;
x = "Hello";
console.log(x);

// Closure example
function outer() {
    let x = "outer value";
    function inner() {
        console.log("Accessing:", x);
    }
    return inner;
}

const f = outer();
f();
